code/modules/mob/living/living.dm

```diff
@@ -968,17 +977,28 @@
 			lying = incapacitated(INCAPACITATION_KNOCKDOWN)
 			canmove = !incapacitated(INCAPACITATION_DISABLED)
 
+	if(lying && (incapacitated(INCAPACITATION_KNOCKOUT) || incapacitated(INCAPACITATION_STUNNED))) // CHOMPAdd - Making sure we're in good condition to crawl
+		canmove = 0
+		drop_both_hands()
+	else
+		canmove = 1
+
 	if(lying)
 		density = FALSE
+	/* CHOMPEdit - Allow us to hold stuff while laying down.
 		if(l_hand)
 			unEquip(l_hand)
 		if(r_hand)
 			unEquip(r_hand)
 		for(var/obj/item/weapon/holder/holder in get_mob_riding_slots())
 			unEquip(holder)
+	*/
 		update_water() // Submerges the mob.
+		stop_pulling() // CHOMPAdd - Stops pulling things
+		pass_flags = PASSTABLE
 	else
 		density = initial(density)
+		pass_flags = 0
```

Comment:
You are changing pass_flags here but many other sources can edit pass_flags.

You're gunna need to store whatever your old passflags were before modifying them

e.g
`pass_flags_precrawl = passflags
passflags |= PASSTABLE
else
pass_flags = pass_flags_precrawl`

Reply:
Bitflags are also appended, not set. There are several flags in one var

`BITFLAG = A FLAG`
will clear all flags and set it to ONLY your flag.

`BITFLAG |= A FLAG`
will add your flag to all the existing flags set in the var. If A FLAG already exists, then there is no change.

It is also possible to ONLY remove ONE flag from a bitflag var, that is achieved by
`BITFLAG &= ~A_FLAG`